streamer/src/nonblocking/quic.rs

```diff
@@ -146,21 +147,45 @@ async fn setup_connection(
                 let stake = *stake;
                 let total_stake = staked_nodes.total_stake;
                 drop(staked_nodes);
-                let mut connection_table_l = staked_connection_table.lock().unwrap();
-                if connection_table_l.total_size >= max_staked_connections {
-                    let max_connections = max_percentage_full.apply_to(max_staked_connections);
-                    let num_pruned = connection_table_l.prune_oldest(max_connections);
-                    stats.num_evictions.fetch_add(num_pruned, Ordering::Relaxed);
-                }
+
                 connection.set_max_concurrent_uni_streams(
                     VarInt::from_u64(
                         ((stake as f64 / total_stake as f64) * QUIC_TOTAL_STAKED_CONCURRENT_STREAMS)
                             as u64,
                     )
                     .unwrap(),
                 );
-                (connection_table_l, stake)
-            } else {
+
+                let mut connection_table_l = staked_connection_table.lock().unwrap();
+                if connection_table_l.total_size >= max_staked_connections {
+                    let mut num_pruned = connection_table_l.prune_random(stake);
+                    if num_pruned == 0 {
+                        if max_unstaked_connections > 0 {
```

Comment:
Can we make checking and pruning the unstaked connection table its own function so that we don't have two copies of that logic?

Reply:
done